packages/next/src/server/app-render/app-render.tsx

```diff
@@ -482,6 +482,18 @@ async function generateDynamicFlightRenderResult(
   })
 }
 
+/**
+ * Crawlers will inadvertently think the canonicalUrl in the RSC payload should be crawled
+ * when our intention is to just seed the router state with the current URL.
+ * This function splits up the pathname so that we can later join it on
+ * when we're ready to consume the path.
+ */
+function prepareInitialCanonicalUrl(url: RequestStore['url']) {
```

Comment:
Are we sure we can never have double slashes in this url? We wouldn’t recover that properly on the client since we would lose the empty segment. 

There really isn’t a reason to split this at all, we just need to obfuscate that it is a url. We could replace every \ with something invalid for a url and the unconvert it on the client. We could also just leave empty segments and let them rejoin. We could also just split the string every n characters and do concat.